doc/specs/#4066 - Theme-controlled color scheme switch.md

```diff
@@ -0,0 +1,79 @@
+---
+author: Ítalo Masserano arkthur/italo.masserano@gmail.com
+created on: 2022-03-02
+last updated: 2022-03-02
+issue id: 4066
+---
+
+# Theme-controlled color scheme switch
+
+## Abstract
+
+The idea is for Windows Terminal to change automatically its color schemes according to what theme is selected, including the case where `system` theme is selected.
+
+## Inspiration
+
+I work remotely as a developer, so I have to spend a lot of hours in front of my PC screen. In my setup, right behind my desk I have a window, which is the only source of natural sunlight in my room.
+
+Normally I like dark modes in all the programs and apps I use, but when there's too much sunlight, it becomes annoying, and sometimes even painful, to work in dark mode. So, I have all the programs and apps I use (at least, those that can) set to switch their color themes to what the system has.
+
+The company I work for sent me a Macbook Pro, and my personal phone is an Android, both with automatic dark mode at sunset and light mode at sunrise, and in those devices it's been working relatively well. In Windows, as it is known, there's no such feature, so I manually change between dark and light mode when it's needed, and most of the programs and apps I use go along with this change. Windows Terminal, is not one of them.
+
+The theme changes just as expected, but in an app like this, this change only affects the top of the window, leaving almost all of the screen at the mercy of what the color scheme is, and it doesn't depend on the theme, which defeats any attempt to make a good use of the `system` theme feature.
+
+## Solution Design
+
+Could be implemented in the form of:
+
+```json
+"colorScheme": {
+    "light": "BlulocoLight",
+    "dark": "BlulocoDark"
+}
+```
+or:
+
+```json
+    "colorSchemeLight": "BlulocoLight",
+    "colorSchemeDark": "BlulocoDark"
+```
+
+## UI/UX Design
+
+In a first version it could look like the terminal in Visual Studio Code, and an improvement could be to have light mode specific color schemes, just like those already present in Windows terminal. A good idea could be to get an inspiration in Dark++ and Light++ VSCode color themes.
+
+A user could benefit from a more healthy light level contrast between the screen their looking at and the environment they are, reducing the risk of headache or developing/intensifying eye problems, and any other related eye conditions. Plus, it adds to a more consistent experience between different programs and apps, and the system itself.
+
```

Comment:
(if this were us)

We might add a section here about what happens in the Settings UI if a user does enable this feature. For now though, we're cool with just replacing the color scheme dropdown on the Profile Appearance page with a text box that says "this setting must be configured in the settings file", if the scheme is set to a set of different schemes. 

We also have half a mind to adjust how schemes are configured via the UI entirely, so we're not gonna ask you to add this bit 😝 This is firmly an "us" problem.

Reply:
I appreciate that last part 😆

Although, I'm on board with all the "this setting must be configured in the settings file" thingy. I think it's wise to introduce such a change as gradually as possible.